mock/src/components/REPLFunction.tsx

```diff
@@ -0,0 +1,15 @@
+import { Dispatch, SetStateAction, useState } from "react";
+import { HistoryEntry } from "./REPL";
+/**
+ * A command-processor function for our REPL. The function returns a string,
+ * which is the value to print to history when the command is done executing.
+ *
+ * The arguments passed in the input (which need not be named "args") should
+ * *NOT* contain the command-name prefix.
+ */
+
```

Comment:
The use of this data structure makes is a great use of a strategy pattern to make the code easily changeble